Add user migration and model

Gmarn opened the pull request:
## Summary

- Create migration for Users;
- Set up models for Users;

## Trello card

https://trello.com/c/4FmHxKxM

Gmarn committed:
Add user migration and model

- Create migration for Users;
- Set up models for Users;

igna92ts commented on app/models/user.js:
```diff
@@ -0,0 +1,22 @@
+    },
+    {}
+  );
+  User.associate = function(models) {
```
this goes in classMathods, and you can return the `define` result directly
Like this:
```
module.exports = (sequelize, DataTypes) => {
  return sequelize.define(
    'user',
    {
      firstName: DataTypes.STRING,
      lastName: DataTypes.STRING,
      email: DataTypes.STRING,
      password: DataTypes.STRING
    },
    {
      classMethods: {
        associate(models) {
          // associations can be defined here
        }
      }
    }
  );
};
```
The other thing is that in postgres, tables wont allow upper case so user lower case names instead in the `define`

Gmarn committed:
Erase upper case

Gmarn committed:
Add services to use console and change some line in model

Gmarn committed:
Sign up end point

- POST to add user to DB;
- Validation on email, pwd, firstName and lastName;
- Password hashing;
- DB error logged in logs;

Gmarn committed:
Test user sign up endpoint

- Some test;

Gmarn committed:
Change some texts in test

Gmarn committed:
Set up sign in

- Route + Middleware + Controller for signIn;
- session manager in service to generate session token;
- test + doc;

Gmarn committed:
Some minor modification

Gmarn committed:
Minor fix

Gmarn committed:
Add some verifictation on test

Gmarn committed:
Merge pull request #6 from wolox-training/sign-in

Set up sign in

Gmarn committed:
Merge branch 'sign-up' into sign-up-test

Gmarn committed:
Merge pull request #5 from wolox-training/sign-up-test

Test user sign up endpoint

Gmarn committed:
Merge pull request #4 from wolox-training/sign-up

Sign up end point
